ambry-rest/src/main/java/com.github.ambry.rest/NettyServer.java

```diff
@@ -49,27 +50,20 @@
   private final NettyMetrics nettyMetrics;
   private final EventLoopGroup bossGroup;
   private final EventLoopGroup workerGroup;
-  private final RestRequestHandler requestHandler;
-  private final PublicAccessLogger publicAccessLogger;
-  private final RestServerState restServerState;
+  private final ArrayList<Map.Entry<String, ChannelHandler>> channelHandlers;
```

Comment:
This looks like a hack. I assume the intention here is a list of pairs? You shouldn't have to use `Map.Entry` for that. There are no keys and values and there are no maps in this context.
I suggest creating a small class of your own with meaningful names for the tuple/pair.